mongomock/collection.py

```diff
@@ -254,6 +256,8 @@ def update(self, spec, document, upsert = False, manipulate = False,
                         existing_document.update(self._internalize_dict(document))
                         if existing_document['_id'] != _id:
```

Comment:
@vmalloc When I was fixing this to handle dict _id's I found that mongo doesn't allow changing `_id` in an update.  It throws this message:

```
The _id field cannot be changed from ...
```

For compatibility, should this throw an exccption instead of delete/insert?